### ISSUE TITLE
TFIDF vectorizer

### ISSUE THREAD

**Issue by dfernan** [dask/dask#6663]
I'd want to be able to do TF-IDF calculations in Dask in a similar fashion as in apache spark. https://spark.apache.org/docs/latest/mllib-feature-extraction.html

Right now, counting/hashing vectorizer are similar and it does exist. They’re in Dask-ML and are the same as TFIDF without the normalization/function.

Another implementation exists in rapids/cuml, but not sure how easy it'd be to translate it to the Dask framework.
https://github.com/rapidsai/cuml/blob/cdb14e7de6a40d8d707d29b2889a89aa553125ee/python/cuml/feature_extraction/_tfidf_vectorizer.py.

**Comment by stsievert** [comment]
I think this issue belongs in Dask-ML. @TomAugspurger is it possible to move this issue over there?

I'm not seeing tfidf normalization in [CountVectorizer][2] or [HashingVectorizer][1]. The documentation of HashingVectorizer specifically mentions the following:

> There are also a couple of cons (vs using a CountVectorizer with an in-memory vocabulary):
>
> * ...
> * no IDF weighting as this would render the transformer stateful.

One method to resolve this issue would be to implement [TfIdfTransformer.fit][4].



[1]:https://ml.dask.org/modules/generated/dask_ml.feature_extraction.text.HashingVectorizer.html
[2]:https://ml.dask.org/modules/generated/dask_ml.feature_extraction.text.CountVectorizer.html
[3]:https://scikit-learn.org/stable/modules/generated/sklearn.feature_extraction.text.TfidfTransformer.html
[4]:https://github.com/scikit-learn/scikit-learn/blob/0fb307bf39bbdacd6ed713c00724f8f871d60370/sklearn/feature_extraction/text.py#L1424
